bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/handler/OpenWebNetThermoregulationHandler.java

```diff
@@ -0,0 +1,375 @@
+        logger.debug("updateModeAndFunction() for thing: {} msg={}", thing.getUID(), tmsg);
+
+        if (tmsg.getWhat() == null) {
+            logger.debug("updateModeAndFunction() Could not parse Mode from: {}", tmsg.getFrameValue());
+            return;
+        }
+
+        try {
+            logger.debug("updateModeAndFunction() WHAT={}", tmsg.getWhat());
+            WHAT w = WHAT.fromValue(tmsg.getWhat().value());
+
+            OPERATION_MODE mode = w.mode();
+            FUNCTION function = w.function();
+
+            if (w == WHAT.HEATING) {
+                function = FUNCTION.HEATING;
+            } else if (w == WHAT.CONDITIONING) {
+                function = FUNCTION.COOLING;
+            }
+
+            if (mode == OPERATION_MODE.MANUAL) {
+                logger.debug("updateModeAndFunction() function={} mode={} setPointTemp={}°", function.toString(),
+                        mode.toString(), currentSetPointTemp);
+            } else {
+                logger.debug("updateModeAndFunction() function={} mode={}", function.toString(), mode.toString());
+            }
+
+            updateState(CHANNEL_MODE, new StringType(mode.toString()));
+            updateState(CHANNEL_FUNCTION, new StringType(function.toString()));
+
+            // store current function
+            currentFunction = function;
+        } catch (Exception e) {
+            logger.warn("updateModeAndFunction() FrameException on frame {}: {}", tmsg, e.getMessage());
+            // do not update channel state, simply skip wrong message
+        }
+    }
+
+    private void updateTemperature(Thermoregulation tmsg) {
+        logger.debug("updateTemperature() for thing: {}", thing.getUID());
+        try {
+            Double temp = Thermoregulation.parseTemperature(tmsg);
+            updateState(CHANNEL_TEMPERATURE, new DecimalType(temp));
+        } catch (FrameException e) {
+            logger.warn("updateTemperature() FrameException on frame {}: {}", tmsg, e.getMessage());
+            updateState(CHANNEL_TEMPERATURE, UnDefType.UNDEF);
+        }
+    }
+
+    private void updateSetpoint(Thermoregulation tmsg) {
+        logger.debug("updateSetpoint() for thing: {}", thing.getUID());
+        String channelID = CHANNEL_TEMP_SETPOINT;
+        try {
+            Double temp = Thermoregulation.parseTemperature(tmsg);
+            updateState(channelID, new DecimalType(temp));
+            // store current setPoint T
+            currentSetPointTemp = temp;
+        } catch (FrameException e) {
+            logger.warn("updateSetpoint() FrameException on frame {}: {}", tmsg, e.getMessage());
+            updateState(channelID, UnDefType.UNDEF);
+        }
+    }
+
+    private void updateFanCoilSpeed(Thermoregulation tmsg) {
+        logger.debug("updateFanCoilSpeed() for thing: {}", thing.getUID());
+
+        try {
+            FAN_COIL_SPEED speed = Thermoregulation.parseFanCoilSpeed(tmsg);
```

Comment:
Types in Java should be CamelCase.
```suggestion
            FanCoilSpeed speed = Thermoregulation.parseFanCoilSpeed(tmsg);
```

Reply:
Ok. FanCoilSpeed, OperationMode, Function, ValveOrActuatorStatus, WhatThermo and DimThermo were updated.